### PR TITLE
✨ Add VideoInterface support to Connatix Player component

### PR DESCRIPTION
**Pull request by cristisilav**
The purpose of this PR is to add support for the VideoInterface API, as a means to enable the `dock` attribute to be used in conjunction with the Connatix Player AMP component.

### CODE REVIEW

**Comment by erwinmombay** [comment]
@cristisilav would you mind rebasing this to the latest from main. looks like the tests are failing due to a node issue. i believe it is resolved on main

**Comment by erwinmombay** [comment]
@banaag do you know why the validator is erroring out here?

they added the a new change to allow `dock` but we are seeing this error https://app.circleci.com/pipelines/github/ampproject/amphtml/29459/workflows/f4cec47d-7c83-4c77-ab58-1fff57930cc5/jobs/630707

**Comment by banaag** [comment]
> @banaag do you know why the validator is erroring out here?
> 
> they added the a new change to allow `dock` but we are seeing this error https://app.circleci.com/pipelines/github/ampproject/amphtml/29459/workflows/f4cec47d-7c83-4c77-ab58-1fff57930cc5/jobs/630707

From a valid example/test using amp video docking:
https://github.com/ampproject/amphtml/blob/4ce3cd79520dbeaf5ed5364cbff58d3d71dee40e/extensions/amp-video-docking/0.1/test/validator-amp-video-docking-amp-brightcove.out#L18

One difference I can detect is this:

For valid (link above):
<script async custom-element="amp-video-docking" src="https://cdn.ampproject.org/v0/amp-video-docking-latest.js"></script>

For this player:
  <script async custom-element="amp-video-docking" src="https://cdn.ampproject.org/v0/amp-video-docking-0.1.js"></script>

**Comment by cristisilav** [comment]
@banaag @erwinmombay Made the changes you suggested and that fixed the validator tests, but it still seems like tests are failing on the example page. Not sure what other avenue I should explore here

**Comment by erwinmombay** [comment]
thanks @cristisilav , ill take a closer look today and recommendation on how to fix. was OOO last week

File: extensions/amp-connatix-player/0.1/test/validator-amp-connatix-player.out [hunk 1]
```diff
@@ -28,6 +28,15 @@ amp-connatix-player/0.1/test/validator-amp-connatix-player.html:23:2 The mandato
 |      width="150"
 |      height="80">
 |    </amp-connatix-player>
+|    <!-- Invalid: dock without amp-video-docking extension -->
+|    <amp-connatix-player
+>>   ^~~~~~~~~
+amp-connatix-player/0.1/test/validator-amp-connatix-player.html:29:2 The attribute 'dock' requires including the 'amp-video-docking' extension JavaScript.
+|      layout="responsive"
```
Comment (erwinmombay) [inline extensions/amp-connatix-player/0.1/test/validator-amp-connatix-player.out:35]: missing `data-player-id` here. 
I'm still looking into the other error, hoping to figure it out by today

File: examples/connatix-player.amp.html [hunk 2]
```diff
@@ -39,6 +40,13 @@ <h3>Connatix Player Responsive - With Media Id</h3>
     width="480"
     height="270">
   </amp-connatix-player>
-
+  <h3>Connatix Player With Docking</h3>
```
Comment (erwinmombay) [inline examples/connatix-player.amp.html:43]: revert the changes in this file for now. looks like this is the issue blocking us.

File: extensions/amp-connatix-player/0.1/test/validator-amp-connatix-player.out (hunk 1 above)
Comment (erwinmombay) [reply extensions/amp-connatix-player/0.1/test/validator-amp-connatix-player.out:35]: can you copy this file exactly https://github.com/ampproject/amphtml/pull/39601/files#diff-632c252f906759e11a7fc6bbc1d08f6832d619b9ac86f61926535c8ea22dad7c

theres a newline required on the end i think.

**Comment by erwinmombay** [comment]
lmk if the latest round of fixes work, i got the validator passing in this forked PR https://github.com/ampproject/amphtml/pull/39601

**Review by erwinmombay — APPROVED** [review]
